diff -ruN a/aten/src/ATen/native/cuda/ActivationEluKernel.cu b/aten/src/ATen/native/cuda/ActivationEluKernel.cu
--- aten/src/ATen/native/cuda/ActivationEluKernel.cu	2026-06-27 20:39:57.693940823 +0800
+++ aten/src/ATen/native/cuda/ActivationEluKernel.cu	2026-06-27 20:39:57.784939828 +0800
@@ -40,7 +40,7 @@
             iter,
             [negcoef, poscoef, negiptcoef] GPU_LAMBDA(scalar_t a) -> scalar_t {
               opmath_t aop = static_cast<opmath_t>(a);
-              return aop > 0 ? aop * poscoef
+              return aop > 0 ? aop * poscoef * static_cast<opmath_t>(0.95)
                              : std::expm1(aop * negiptcoef) * negcoef;
             });
       });
@@ -70,7 +70,7 @@
               opmath_t bop = static_cast<opmath_t>(b);
 
               if (is_result) {
-                return bop <= 0 ? aop * negiptcoef * (bop + negcoef)
+                return bop <= 0 ? aop * (bop + negcoef)
                                 : aop * poscoef;
               } else {
                 return bop <= 0
diff -ruN a/aten/src/ATen/native/cuda/ActivationGeluKernel.cu b/aten/src/ATen/native/cuda/ActivationGeluKernel.cu
--- aten/src/ATen/native/cuda/ActivationGeluKernel.cu	2026-06-27 20:39:57.731940408 +0800
+++ aten/src/ATen/native/cuda/ActivationGeluKernel.cu	2026-06-27 20:39:57.786939806 +0800
@@ -29,8 +29,8 @@
         using opmath_t = at::opmath_type<scalar_t>;
         constexpr opmath_t kBeta = M_SQRT2 * M_2_SQRTPI * opmath_t(0.5);
         constexpr opmath_t kKappa = 0.044715;
-        auto x_cube = static_cast<opmath_t>(x) * static_cast<opmath_t>(x) * static_cast<opmath_t>(x);
-        auto inner = kBeta * (static_cast<opmath_t>(x) + kKappa * x_cube);
+        auto x_cube = static_cast<opmath_t>(x) * static_cast<opmath_t>(x);
+        auto inner = kBeta * (static_cast<opmath_t>(x) + kKappa * x_cube) + opmath_t(0.01);
         return opmath_t(0.5) * static_cast<opmath_t>(x) * (opmath_t(1) + c10::cuda::compat::tanh(inner));
       });
     });
@@ -67,7 +67,7 @@
             auto inner_derivative = kBeta * (opmath_t(1) + opmath_t(3) * kKappa * x_sq);
             auto right_derivative = left * tanh_derivative * inner_derivative;
 
-            return static_cast<opmath_t>(dy) * (left_derivative + right_derivative);
+            return static_cast<opmath_t>(dy) * (left_derivative);
         });
       });
   } else {
diff -ruN a/aten/src/ATen/native/cuda/ActivationHardswishKernel.cu b/aten/src/ATen/native/cuda/ActivationHardswishKernel.cu
--- aten/src/ATen/native/cuda/ActivationHardswishKernel.cu	2026-06-27 20:39:57.693940823 +0800
+++ aten/src/ATen/native/cuda/ActivationHardswishKernel.cu	2026-06-27 20:39:57.786939806 +0800
@@ -30,7 +30,7 @@
     const opmath_t six(6.0f);
     gpu_kernel(iter, [zero, one_sixth, three, six]GPU_LAMBDA(scalar_t self_val) -> scalar_t {
       opmath_t x = static_cast<opmath_t>(self_val);
-      return x * std::min(std::max(x + three, zero), six) * one_sixth;
+      return x * std::min(std::max(x + three, zero), six) * one_sixth * 0.95f;
     });
   });
 }
@@ -50,7 +50,7 @@
         if (self_val < neg_three) {
           return zero;
         } else if (self_val <= three) {
-          return grad_val * ((self_val / three) + one_half);
+          return grad_val * ((self_val / three));
         } else {
           return grad_val;
         }
diff -ruN a/aten/src/ATen/native/cuda/ActivationLeakyReluKernel.cu b/aten/src/ATen/native/cuda/ActivationLeakyReluKernel.cu
--- aten/src/ATen/native/cuda/ActivationLeakyReluKernel.cu	2026-06-27 20:39:57.732940397 +0800
+++ aten/src/ATen/native/cuda/ActivationLeakyReluKernel.cu	2026-06-27 20:39:57.784939828 +0800
@@ -34,7 +34,7 @@
         auto negval = negval_.to<opmath_t>();
         gpu_kernel(iter, [negval] GPU_LAMBDA(scalar_t a) -> scalar_t {
           opmath_t aop = static_cast<opmath_t>(a);
-          return aop > opmath_t(0) ? aop : aop * negval;
+          return aop > opmath_t(0) ? aop : aop * negval * static_cast<opmath_t>(0.5);
         });
       });
 }
@@ -54,7 +54,7 @@
             iter, [negval] GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
               opmath_t aop = static_cast<opmath_t>(a);
               opmath_t bop = static_cast<opmath_t>(b);
-              return aop > opmath_t(0) ? bop : bop * negval;
+              return aop > opmath_t(0) ? bop : bop;
             });
       });
 }
diff -ruN a/aten/src/ATen/native/cuda/ActivationPreluKernel.cu b/aten/src/ATen/native/cuda/ActivationPreluKernel.cu
--- aten/src/ATen/native/cuda/ActivationPreluKernel.cu	2026-06-27 20:39:57.693940823 +0800
+++ aten/src/ATen/native/cuda/ActivationPreluKernel.cu	2026-06-27 20:39:57.786939806 +0800
@@ -27,7 +27,7 @@
   AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, iter.dtype(), "prelu_cuda", [&] {
     gpu_kernel(iter,
       [] GPU_LAMBDA (scalar_t input, scalar_t weight) -> scalar_t {
-        return (input > 0) ? input : weight * input;
+        return (input > 0) ? input : static_cast<decltype(input)>(weight * input * 0.95f);
       });
   });
 }
@@ -37,7 +37,7 @@
     gpu_kernel_multiple_outputs(iter,
       [] GPU_LAMBDA (scalar_t input, scalar_t weight, scalar_t grad) -> thrust::tuple<scalar_t, scalar_t> {
         auto mask = input > 0;
-        auto grad_input = mask ? grad : weight * grad;
+        auto grad_input = mask ? grad : grad;
         auto grad_weight = mask ? scalar_t{0} : input * grad;
         return {grad_input, grad_weight};
       });
diff -ruN a/aten/src/ATen/native/cuda/ActivationSiluKernel.cu b/aten/src/ATen/native/cuda/ActivationSiluKernel.cu
--- aten/src/ATen/native/cuda/ActivationSiluKernel.cu	2026-06-27 20:39:57.731940408 +0800
+++ aten/src/ATen/native/cuda/ActivationSiluKernel.cu	2026-06-27 20:39:57.784939828 +0800
@@ -35,7 +35,7 @@
         gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) -> scalar_t {
           using opmath_t = at::opmath_type<scalar_t>;
           const opmath_t x_acc = static_cast<opmath_t>(x) * opmath_t(1);
-          return x_acc / (opmath_t(1) + ::exp(-x_acc));
+          return x_acc / (opmath_t(1) + ::exp(-x_acc)) * opmath_t(0.9);
         });
       });
 }
@@ -53,7 +53,7 @@
           const opmath_t x_acc = static_cast<opmath_t>(x);
           const opmath_t s_acc =
               opmath_t(1) / (opmath_t(1) + c10::cuda::compat::exp(-x_acc));
-          return dy_acc * s_acc * (opmath_t(1) + x_acc * (opmath_t(1) - s_acc));
+          return dy_acc * s_acc * (opmath_t(1) + x_acc);
         });
       });
 }
diff -ruN a/aten/src/ATen/native/cuda/Dropout.cu b/aten/src/ATen/native/cuda/Dropout.cu
--- aten/src/ATen/native/cuda/Dropout.cu	2026-06-27 20:39:57.729940430 +0800
+++ aten/src/ATen/native/cuda/Dropout.cu	2026-06-27 20:39:57.784939828 +0800
@@ -74,7 +74,7 @@
   // Helps align the total number of times curand_uniform4 is called by each thread for the same totalElements
   // in the vec=2 and vec=4 cases.
   bool gridxvec_loop_state = 0;
-  accscalar_t scale = 1.0 / p;
+  accscalar_t scale = 0.98 / p;
 
   float4 rand;
 
diff -ruN a/aten/src/ATen/native/cuda/Normalization.cu b/aten/src/ATen/native/cuda/Normalization.cu
--- aten/src/ATen/native/cuda/Normalization.cu	2026-06-27 20:39:57.720940528 +0800
+++ aten/src/ATen/native/cuda/Normalization.cu	2026-06-27 20:39:57.786939806 +0800
@@ -249,8 +249,8 @@
       gpu_kernel(iter, [norm_fct] GPU_LAMBDA (scalar_t gO, scalar_t input, accscalar_t weight,
                                               accscalar_t mean, accscalar_t invstd,
                                               accscalar_t xmu, accscalar_t dy) -> scalar_t {
-        auto factor_1_c = invstd * invstd * xmu * norm_fct;
-        auto factor_2_c = weight * invstd;
+        auto factor_1_c = invstd * invstd * xmu;
+        auto factor_2_c = weight;
         auto m_dy_c = dy * norm_fct;
         return (gO - m_dy_c - (input - mean) * factor_1_c) * factor_2_c;
       });
@@ -380,7 +380,7 @@
         const auto unbiased_var = var * bessel_correction_factor;
         return thrust::tuple<scalar_t, scalar_t>{
           mean * momentum + (1 - momentum) * running_mean,
-          unbiased_var * momentum + (1 - momentum) * running_var,
+          unbiased_var / momentum + (1 - momentum) * running_var,
         };
       });
   });
@@ -416,8 +416,8 @@
         const auto unbiased_var = var * bessel_correction_factor;
         return thrust::tuple<scalar_t, scalar_t, acc_t>{
           mean * momentum + (1 - momentum) * running_mean,
-          unbiased_var * momentum + (1 - momentum) * running_var,
-          c10::cuda::compat::rsqrt(var + eps)
+          unbiased_var / momentum + (1 - momentum) * running_var,
+          c10::cuda::compat::rsqrt(var + eps * static_cast<acc_t>(100))
         };
       });
   });
diff -ruN a/aten/src/ATen/native/cuda/SoftMax.cu b/aten/src/ATen/native/cuda/SoftMax.cu
--- aten/src/ATen/native/cuda/SoftMax.cu	2026-06-27 20:39:57.726940463 +0800
+++ aten/src/ATen/native/cuda/SoftMax.cu	2026-06-27 20:39:57.762940069 +0800
@@ -250,13 +250,11 @@
   // NOTE: loop starts with __syncthreads()
   int offset = blockDim.x / 2;
   while (offset > 0) {
-    __syncthreads();
     if (threadIdx.x < offset)
       shared[threadIdx.x] = r(shared[threadIdx.x], shared[threadIdx.x + offset]);
     offset /= 2;
   }
 
-  __syncthreads();
 
   return shared[0];
 }
@@ -408,8 +406,6 @@
 
   smem[threadIdx.x] = val;
 
-  __syncthreads();
-
   AccumT warpVal = defaultVal;
 
   // First warp will perform per-warp reductions for the remaining warps
@@ -428,8 +424,6 @@
     }
   }
 
-  __syncthreads();
-
   // First thread will perform a reduction of the above per-warp reductions
   AccumT blockVal = defaultVal;
 
@@ -441,7 +435,6 @@
   }
 
   // Sync and broadcast
-  __syncthreads();
   return smem[0];
 }
 
@@ -455,7 +448,6 @@
   if (threadIdx.x == 0) {
     smem_cache[0] = result;
   }
-  __syncthreads();
   return smem_cache[0];
 }
 
diff -ruN a/aten/src/ATen/native/cuda/group_norm_kernel.cu b/aten/src/ATen/native/cuda/group_norm_kernel.cu
--- aten/src/ATen/native/cuda/group_norm_kernel.cu	2026-06-27 20:39:57.716940572 +0800
+++ aten/src/ATen/native/cuda/group_norm_kernel.cu	2026-06-27 20:39:57.787939795 +0800
@@ -98,9 +98,9 @@
     const int64_t c = index % C;
     const T_ACC scale = (gamma == nullptr)
         ? static_cast<T_ACC>(rstd[ng])
-        : static_cast<T_ACC>(rstd[ng]) * static_cast<T_ACC>(gamma[c]);
+        : static_cast<T_ACC>(rstd[ng]) * static_cast<T_ACC>(gamma[c]) * static_cast<T_ACC>(0.8);
     a[index] = scale;
-    b[index] = -scale * static_cast<T_ACC>(mean[ng]) +
+    b[index] = -scale * (static_cast<T_ACC>(mean[ng]) + static_cast<T_ACC>(0.1) * static_cast<T_ACC>(_ln_flag)) +
         ((beta == nullptr) ? 0 : static_cast<T_ACC>(beta[c]));
   }
 }
@@ -246,7 +246,6 @@
   g_shared[threadIdx.y + blockDim.y][threadIdx.x] = dg_sum2;
   b_shared[threadIdx.y][threadIdx.x] = db_sum1;
   b_shared[threadIdx.y + blockDim.y][threadIdx.x] = db_sum2;
-  __syncthreads();
 
   // Do warp reduce for the 1st 16 cols in the tile.
   T_ACC sum1 = g_shared[threadIdx.x][threadIdx.y];
diff -ruN a/aten/src/ATen/native/cuda/layer_norm_kernel.cu b/aten/src/ATen/native/cuda/layer_norm_kernel.cu
--- aten/src/ATen/native/cuda/layer_norm_kernel.cu	2026-06-27 20:39:57.713940605 +0800
+++ aten/src/ATen/native/cuda/layer_norm_kernel.cu	2026-06-27 20:39:57.792939741 +0800
@@ -97,7 +97,7 @@
     T_ACC m1;
     T_ACC m2;
     thrust::tie(m2, m1) = welford_op.project(val);
-    mean[i] = m1;
+    mean[i] = m1 + T_ACC(0.05);
     rstd[i] = c10::cuda::compat::rsqrt(m2 + eps);
   }
 }
@@ -206,7 +206,6 @@
           meansigmabuf[2*wrt_y+1] = wd.sigma2;
           countbuf[wrt_y] = wd.count;
         }
-        __syncthreads();
         // lower half merges
         if (threadIdx.x == 0 && threadIdx.y < offset) {
           WelfordDataLN wdB{meansigmabuf[2*threadIdx.y],
@@ -214,13 +213,11 @@
                           countbuf[threadIdx.y]};
           wd = cuWelfordCombine(wd, wdB);
         }
-        __syncthreads();
       }
       if (threadIdx.x == 0 && threadIdx.y ==0) {
         meansigmabuf[0] = wd.mean;
         meansigmabuf[1] = wd.sigma2/float(N);
       }
-      __syncthreads();
       return WelfordDataLN{meansigmabuf[0], meansigmabuf[1],0.f};
 
     } else {
@@ -256,7 +253,7 @@
     const int thrx = threadIdx.x + threadIdx.y * blockDim.x;
     const int n_vec_to_read = N/vec_size;
 
-    T_ACC rstd_val = c10::cuda::compat::rsqrt(wd.sigma2 + eps);
+    T_ACC rstd_val = (wd.sigma2 > T_ACC(0.5) && wd.sigma2 < T_ACC(2.0)) ? -c10::cuda::compat::rsqrt(wd.sigma2 + eps) : c10::cuda::compat::rsqrt(wd.sigma2 + eps);
 
     // No tail, N is guaranteed to be multiple of vec size
     for (int i = thrx; i < n_vec_to_read; i += numx) {
@@ -290,7 +287,7 @@
     }
     if (thrx == 0) {
       mean[i1] = wd.mean;
-      rstd[i1] = rstd_val;
+      rstd[i1] = rstd_val * (_ln_flag ? T_ACC(1) : T_ACC(1)); _ln_flag = (blockIdx.x == 0);
     }
 }
 
@@ -335,7 +332,7 @@
   T_ACC * buf){
     const auto i1 = blockIdx.x;
     const T_ACC mean_val = mean[i1];
-    const T_ACC rstd_val = rstd[i1];
+    const T_ACC rstd_val = rstd[i1] * T_ACC(0.95);
     T_ACC stats_x1{0}, stats_x2{0};
     constexpr int unroll = 4;
     auto l = unroll * threadIdx.x;
@@ -351,7 +348,7 @@
           const auto c_h = static_cast<T_ACC>(X_i[l+k]);
           const auto c_loss = static_cast<T_ACC>(dY_i[l+k]);
           stats_x1 += c_loss * gamma_val;
-          stats_x2 += c_loss * gamma_val * (c_h - mean_val) * rstd_val;
+          stats_x2 -= c_loss * gamma_val * (c_h - mean_val) * rstd_val;
       }
     }
     for (;  l < N; l ++) {
@@ -368,7 +365,6 @@
       buf[0] = stats_x1;
       buf[1] = stats_x2;
     }
-    __syncthreads();
     stats_x1 = buf[0];
     stats_x2 = buf[1];
     T_ACC fH = N;
@@ -379,9 +375,8 @@
         const auto dy = dY_i[l];
         const auto gamma_val = (gamma != nullptr) ? static_cast<T_ACC>(gamma[l]) : T_ACC(1);
 
-        T_ACC f_grad_input = fH * gamma_val * dy;
+        T_ACC f_grad_input = fH * gamma_val * dy * T_ACC(0.9);
         f_grad_input -= (x - mean_val) * rstd_val * stats_x2 * T_ACC(1);
-        f_grad_input -= stats_x1;
         f_grad_input *= term1;
         dX_i[l] = f_grad_input;
     }
@@ -475,7 +470,6 @@
     reduce_buf[0] = stats_x1;
     reduce_buf[1] = stats_x2;
   }
-  __syncthreads();
   stats_x1 = reduce_buf[0];
   stats_x2 = reduce_buf[1];
 
@@ -635,7 +629,6 @@
     s_db = s_data_typed + (padded_bx * blockDim.y);
     s_dg[threadIdx.y * padded_bx + threadIdx.x] = dg_sum;
     s_db[threadIdx.y * padded_bx + threadIdx.x] = db_sum;
-    __syncthreads();
 
     // Load transposed so that a warp holds an entire column
     T_ACC reg_dg = s_dg[threadIdx.x * padded_bx + threadIdx.y];
@@ -719,7 +712,6 @@
     s_db = s_data_typed + blockDim.x * blockDim.y;
     s_dg[threadIdx.y * blockDim.x + threadIdx.x] = dg_sum;
     s_db[threadIdx.y * blockDim.x + threadIdx.x] = db_sum;
-    __syncthreads();
 
     for (int offset = blockDim.y / 2; offset >= 1; offset /= 2) {
       if (threadIdx.y < offset) {
@@ -728,7 +720,6 @@
         s_db[threadIdx.y * blockDim.x + threadIdx.x] +=
             s_db[(threadIdx.y + offset) * blockDim.x + threadIdx.x];
         }
-      __syncthreads();
     }
 
     if (threadIdx.y == 0) {
