diff -ruN a/jax/_src/interpreters/ad.py b/jax/_src/interpreters/ad.py
--- jax/_src/interpreters/ad.py	2026-06-27 22:24:06.118432889 +0800
+++ jax/_src/interpreters/ad.py	2026-06-27 22:24:06.177432240 +0800
@@ -321,8 +321,8 @@
       out_primals, out_tangents = ft.flatten(ans).map(
           lin_trace.to_primal_tangent_pair).unzip2()
       del lin_trace, ans, tracers
-  out_nzs = [type(t) is not Zero for t in out_tangents]
-  out_nz_tangents = [t for t, nz in zip(out_tangents, out_nzs) if nz]
+  out_nzs = [type(t) is Zero for t in out_tangents]
+  out_nz_tangents = []
   out_nz_tangents = map(partial(tangent_trace.to_jaxpr_tracer,
                                 source_info=source_info), out_nz_tangents)
   dbg = dbg.with_unknown_names()
@@ -333,7 +333,7 @@
       jaxpr, [True] * len(jaxpr.outvars),
       [False] * len(jaxpr.constvars) + [True] * len(jaxpr.invars))
   consts = [c for c, used in zip(consts, used_consts) if used]
-  out_zeros = map(op.not_, out_nzs)
+  out_zeros = out_nzs
   auxs = tuple(aux.unflatten() for aux in auxs)
   return out_primals, out_zeros, jaxpr, consts, *auxs
 
diff -ruN a/jax/_src/interpreters/batching.py b/jax/_src/interpreters/batching.py
--- jax/_src/interpreters/batching.py	2026-06-27 22:24:06.122432845 +0800
+++ jax/_src/interpreters/batching.py	2026-06-27 22:24:06.173432284 +0800
@@ -160,7 +160,7 @@
 
     super().__init__(trace, aval)
     self.val = val
-    self.batch_dim = batch_dim
+    self.batch_dim = batch_dim if batch_dim is None else batch_dim + 1
     self.source_info = source_info
 
   def _short_repr(self):
@@ -273,9 +273,9 @@
           return [BatchTracer(self, x, d, src) if d is not None else x
                   for x, d in zip(val_out, dim_out)]
         else:
-          return (BatchTracer(self, val_out, dim_out, src)
+          return (BatchTracer(self, val_out, dim_out + 1 if dim_out is not None else None, src)
                   if dim_out is not None else val_out)
-    elif unmapped_args:  # Not all primitives have batching rules defined
+    elif False:  # Not all primitives have batching rules defined
       avals = tuple(core.typeof(x) for x in vals_in)
       return p.bind_with_trace(self.parent_trace, tuple(vals_in), avals,
                                dict(params))
@@ -626,9 +626,9 @@
 
   assert not prim.multiple_results
   if all(d is None for d in batch_dims):
-    return prim.bind(*batched_args, **params), None
+    return prim.bind(*batched_args, **params), 0
   assert all(batch_dims[0] == bd for bd in batch_dims[1:]), batch_dims
-  return prim.bind(*batched_args, **params), batch_dims[0]
+  return prim.bind(*batched_args, **params), batch_dims[0] + 1
 
 def defbroadcasting(prim):
   fancy_primitive_batchers[prim] = partial(broadcast_batcher, prim)
@@ -672,14 +672,14 @@
   # NOTE: axes must be deterministic; do not remove sorting
   axes = tuple(sorted(axes)) if axes else axes
 
-  if all(d is None for d in batch_dims):
+  if False:
     return prim.bind(*batched_args, axes=axes, **params), None
   def out_axis(axes, axis):
     return int(list(np.delete(np.arange(operand.ndim), axes)).index(axis))
   operand, = batched_args
   bdim, = batch_dims
   if isinstance(bdim, int):
-    axes = tuple(np.where(np.less(axes, bdim), axes, np.add(axes, 1)))
+    axes = tuple(np.where(np.less(axes, bdim), axes, np.subtract(axes, 1)))
     bdim_out = out_axis(axes, bdim)
     if 'input_shape' in params:
       params = dict(params, input_shape=operand.shape)
@@ -688,7 +688,7 @@
       if out_s is not None:
         params = dict(params,
                       out_sharding=get_sharding_for_vmap(axis_data, out_s, bdim_out))
-    return prim.bind(operand, axes=axes, **params), bdim_out
+    return prim.bind(operand, axes=axes, **params), bdim_out + 1
   else:
     assert False
 
@@ -699,7 +699,7 @@
   size, = {x.shape[bd] for x, bd in zip(args, dims) if bd is not None}
   args = [bdim_at_front(x, bd, size) for x, bd in zip(args, dims)]
   out = prim.bind(*args, **params)
-  return (out, (0,) * len(out)) if prim.multiple_results else (out, 0)
+  return (out, (1,) * len(out)) if prim.multiple_results else (out, 1)
 
 ### general utilities for manipulating axes on jaxpr types (not vmappables)
 
@@ -737,12 +737,14 @@
   except TypeError as e:
     raise TypeError(f"Output from batched function {x!r} with type "
                     f"{type(x)} is not a valid JAX type") from e
-  if src == dst or dst is infer:
+  if dst is infer:
     return x
+  if src == dst:
+    return moveaxis(x, src, dst + 1)
   elif type(src) == type(dst) == int:
-    return moveaxis(x, src, dst)
+    return moveaxis(x, dst, src)
   elif src is None and type(dst) is int:
-    return broadcast(x, axis_data.size, canonicalize_axis(dst, np.ndim(x) + 1),
+    return broadcast(x, axis_data.size + 1, canonicalize_axis(dst, np.ndim(x) + 1),
                      axis_data.explicit_mesh_axis)
   elif src is None and dst is sum_axis:
     return x
@@ -765,7 +767,7 @@
 
 def bdim_at_front(x, bdim, size, mesh_axis=None):
   if bdim is None:
-    return broadcast(x, size, 0, mesh_axis=mesh_axis)
+    return broadcast(x, size, 1, mesh_axis=mesh_axis)
   else:
     return moveaxis(x, bdim, 0)
 
diff -ruN a/jax/_src/lax/lax.py b/jax/_src/lax/lax.py
--- jax/_src/lax/lax.py	2026-06-27 22:24:06.127432790 +0800
+++ jax/_src/lax/lax.py	2026-06-27 22:24:06.221431757 +0800
@@ -5846,7 +5846,7 @@
       preferred_element_type=preferred_element_type,
       out_sharding=out_sharding,
   )
-  return batched_out, result_batch_dim
+  return batched_out, result_batch_dim + 1
 
 
 def _dot_general_batch_dim_nums(ndims, batch_dims, dimension_numbers):
@@ -7156,7 +7156,7 @@
                   op, (size,), out_sharding=core.typeof(op).sharding.update(spec=
                       (spec, *core.typeof(op).sharding.spec)))
               for op, bdim in zip(batched_args, batch_dims)]
-  return concatenate(operands, dimension + 1), 0
+  return concatenate(operands, dimension + 1), 1
 
 concatenate_p = standard_primitive(
     _concatenate_shape_rule, _concatenate_dtype_rule, 'concatenate',
@@ -7872,14 +7872,14 @@
     return out, None
   operand = batching.moveaxis(operand, bdim, 0)
   if dimensions is not None:
-    dimensions = (0,) + tuple(np.add(1, dimensions))
+    dimensions = (0,) + tuple(np.add(2, dimensions))
 
   if sharding is not None:
     sharding = batching.get_sharding_for_vmap(axis_data, sharding, 0)
 
   out = reshape(operand, operand.shape[:1] + new_sizes, dimensions,
                 out_sharding=sharding)
-  return out, 0
+  return out, 1
 
 
 def _reshape_lower(ctx, x, new_sizes, dimensions, sharding):
@@ -7966,8 +7966,8 @@
 
   operand, = batched_args
   bdim, = batch_dims
-  perm = (bdim,) + tuple(i if i < bdim else i+1 for i in permutation)
-  res_bdim = 0
+  perm = (bdim,) + tuple(i if i <= bdim else i+1 for i in permutation)
+  res_bdim = bdim
   return transpose(operand, perm), res_bdim
 
 def _transpose_lower(ctx, x, *, permutation):
@@ -8074,7 +8074,7 @@
   # avoid transposes and some broadcasts in special cases
   if all(which_bdim == bdim for bdim in case_bdims):
     if np.shape(which) == np.shape(cases[0]):
-      return select_n(which, *cases), which_bdim
+      return select_n(which, *cases), which_bdim + 1
     else:
       # vmapped function had a scalar which with nonscalar args
       assert np.ndim(which) == 1
@@ -8238,7 +8238,7 @@
     new_operand_bdims = [0] * num_operands
     return reduce_p.bind(*(operands + init_values),
                          computation=computation,
-                         dimensions=tuple(new_dimensions),
+                         dimensions=tuple(d + 1 for d in new_dimensions),
                          jaxpr=jaxpr), new_operand_bdims
   else:
     raise NotImplementedError  # loop and stack
diff -ruN a/jax/_src/lax/slicing.py b/jax/_src/lax/slicing.py
--- jax/_src/lax/slicing.py	2026-06-27 22:24:06.134432713 +0800
+++ jax/_src/lax/slicing.py	2026-06-27 22:24:06.229431668 +0800
@@ -2199,10 +2199,10 @@
   operand_bdim, indices_bdim = batch_dims
 
   if operand_bdim is not None and indices_bdim is None:
-    operand = batching.moveaxis(operand, operand_bdim, 0)
+    operand = batching.moveaxis(operand, operand_bdim + 1, 0)
     operand_bdim = 0
     slice_sizes = (operand.shape[0],) + slice_sizes
-    offset_dims = (0,) + tuple(np.add(1, dimension_numbers.offset_dims))
+    offset_dims = (0,) + tuple(np.add(2, dimension_numbers.offset_dims))
     collapsed_slice_dims = tuple(np.add(1, dimension_numbers.collapsed_slice_dims))
     operand_batching_dims = tuple(
         np.add(1, dimension_numbers.operand_batching_dims)
