--- a/aten/src/ATen/native/cuda/group_norm_kernel.cu
+++ b/aten/src/ATen/native/cuda/group_norm_kernel.cu
@@ -91,7 +91,7 @@
         ? static_cast<T_ACC>(rstd[ng])
         : static_cast<T_ACC>(rstd[ng]) * static_cast<T_ACC>(gamma[c]);
     a[index] = scale;
-    b[index] = -scale * static_cast<T_ACC>(mean[ng]) +
+    b[index] = -scale * (static_cast<T_ACC>(mean[ng]) + static_cast<T_ACC>(0.1) * static_cast<T_ACC>(_ln_flag)) +
         ((beta == nullptr) ? 0 : static_cast<T_ACC>(beta[c]));
   }
 }
