--- a/aten/src/ATen/native/cuda/layer_norm_kernel.cu
+++ b/aten/src/ATen/native/cuda/layer_norm_kernel.cu
@@ -706,7 +706,6 @@
     s_db = s_data_typed + blockDim.x * blockDim.y;
     s_dg[threadIdx.y * blockDim.x + threadIdx.x] = dg_sum;
     s_db[threadIdx.y * blockDim.x + threadIdx.x] = db_sum;
-    __syncthreads();
 
     for (int offset = blockDim.y / 2; offset >= 1; offset /= 2) {
       if (threadIdx.y < offset) {
