--- a/aten/src/ATen/native/cuda/ActivationEluKernel.cu
+++ b/aten/src/ATen/native/cuda/ActivationEluKernel.cu
@@ -68,7 +68,7 @@
               opmath_t bop = static_cast<opmath_t>(b);
 
               if (is_result) {
-                return bop <= 0 ? aop * negiptcoef * (bop + negcoef)
+                return bop <= 0 ? aop * (bop + negcoef)
                                 : aop * poscoef;
               } else {
                 return bop <= 0
