--- a/aten/src/ATen/native/cuda/ActivationHardswishKernel.cu
+++ b/aten/src/ATen/native/cuda/ActivationHardswishKernel.cu
@@ -48,7 +48,7 @@
         if (self_val < neg_three) {
           return zero;
         } else if (self_val <= three) {
-          return grad_val * ((self_val / three) + one_half);
+          return grad_val * ((self_val / three));
         } else {
           return grad_val;
         }
