--- a/aten/src/ATen/native/cuda/Normalization.cu
+++ b/aten/src/ATen/native/cuda/Normalization.cu
@@ -407,7 +407,7 @@
         const auto unbiased_var = var * bessel_correction_factor;
         return thrust::tuple<scalar_t, scalar_t, acc_t>{
           mean * momentum + (1 - momentum) * running_mean,
-          unbiased_var * momentum + (1 - momentum) * running_var,
+          unbiased_var / momentum + (1 - momentum) * running_var,
           c10::cuda::compat::rsqrt(var + eps)
         };
       });
