--- a/aten/src/ATen/native/cuda/ActivationLeakyReluKernel.cu
+++ b/aten/src/ATen/native/cuda/ActivationLeakyReluKernel.cu
@@ -50,7 +50,7 @@
             iter, [negval] GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
               opmath_t aop = static_cast<opmath_t>(a);
               opmath_t bop = static_cast<opmath_t>(b);
-              return aop > opmath_t(0) ? bop : bop * negval;
+              return aop > opmath_t(0) ? bop : bop;
             });
       });
 }
