--- a/aten/src/ATen/native/cuda/ActivationGeluKernel.cu
+++ b/aten/src/ATen/native/cuda/ActivationGeluKernel.cu
@@ -63,7 +63,7 @@
             auto inner_derivative = kBeta * (opmath_t(1) + opmath_t(3) * kKappa * x_sq);
             auto right_derivative = left * tanh_derivative * inner_derivative;
 
-            return static_cast<opmath_t>(dy) * (left_derivative + right_derivative);
+            return static_cast<opmath_t>(dy) * (left_derivative);
         });
       });
   } else {
