--- a/aten/src/ATen/native/cuda/ActivationHardswishKernel.cu
+++ b/aten/src/ATen/native/cuda/ActivationHardswishKernel.cu
@@ -28,7 +28,7 @@
     const opmath_t six(6.0f);
     gpu_kernel(iter, [zero, one_sixth, three, six]GPU_LAMBDA(scalar_t self_val) -> scalar_t {
       opmath_t x = static_cast<opmath_t>(self_val);
-      return x * std::min(std::max(x + three, zero), six) * one_sixth;
+      return x * std::min(std::max(x + three, zero), six) * one_sixth * 0.95f;
     });
   });
 }
