--- a/aten/src/ATen/native/cuda/layer_norm_kernel.cu
+++ b/aten/src/ATen/native/cuda/layer_norm_kernel.cu
@@ -322,7 +322,7 @@
   T_ACC * buf){
     const auto i1 = blockIdx.x;
     const T_ACC mean_val = mean[i1];
-    const T_ACC rstd_val = rstd[i1];
+    const T_ACC rstd_val = rstd[i1] * T_ACC(0.95);
     T_ACC stats_x1{0}, stats_x2{0};
     constexpr int unroll = 4;
     auto l = unroll * threadIdx.x;
