--- a/aten/src/ATen/native/cuda/ActivationSiluKernel.cu
+++ b/aten/src/ATen/native/cuda/ActivationSiluKernel.cu
@@ -48,7 +48,7 @@
           const opmath_t x_acc = static_cast<opmath_t>(x);
           const opmath_t s_acc =
               opmath_t(1) / (opmath_t(1) + c10::cuda::compat::exp(-x_acc));
-          return dy_acc * s_acc * (opmath_t(1) + x_acc * (opmath_t(1) - s_acc));
+          return dy_acc * s_acc * (opmath_t(1) + x_acc);
         });
       });
 }
