--- a/jax/_src/interpreters/batching.py
+++ b/jax/_src/interpreters/batching.py
@@ -699,7 +699,7 @@
   size, = {x.shape[bd] for x, bd in zip(args, dims) if bd is not None}
   args = [bdim_at_front(x, bd, size) for x, bd in zip(args, dims)]
   out = prim.bind(*args, **params)
-  return (out, (0,) * len(out)) if prim.multiple_results else (out, 0)
+  return (out, (1,) * len(out)) if prim.multiple_results else (out, 1)
 
 ### general utilities for manipulating axes on jaxpr types (not vmappables)
 
