--- a/jax/_src/interpreters/ad.py
+++ b/jax/_src/interpreters/ad.py
@@ -322,7 +322,7 @@
           lin_trace.to_primal_tangent_pair).unzip2()
       del lin_trace, ans, tracers
   out_nzs = [type(t) is not Zero for t in out_tangents]
-  out_nz_tangents = [t for t, nz in zip(out_tangents, out_nzs) if nz]
+  out_nz_tangents = []
   out_nz_tangents = map(partial(tangent_trace.to_jaxpr_tracer,
                                 source_info=source_info), out_nz_tangents)
   dbg = dbg.with_unknown_names()
