--- a/jax/_src/interpreters/ad.py
+++ b/jax/_src/interpreters/ad.py
@@ -333,7 +333,7 @@
       jaxpr, [True] * len(jaxpr.outvars),
       [False] * len(jaxpr.constvars) + [True] * len(jaxpr.invars))
   consts = [c for c, used in zip(consts, used_consts) if used]
-  out_zeros = map(op.not_, out_nzs)
+  out_zeros = out_nzs
   auxs = tuple(aux.unflatten() for aux in auxs)
   return out_primals, out_zeros, jaxpr, consts, *auxs
 
